cpp/ql/lib/experimental/semmle/code/cpp/ir/dataflow/internal/DataFlowImpl.qll

```diff
@@ -1169,8 +1169,23 @@ pragma[nomagic]
 private int branch(NodeEx n1, Configuration conf) {
   result =
     strictcount(NodeEx n |
-      flowOutOfCallNodeCand1(_, n1, _, n, conf) or flowIntoCallNodeCand1(_, n1, n, conf)
-    )
+        flowOutOfCallNodeCand1(_, n1, _, n, conf) or flowIntoCallNodeCand1(_, n1, n, conf)
+      ) + sum(getLanguageSpecificFlowIntoCallNodeCand1(_, n1, conf))
```

Comment:
I'm not too worried about this. If this predicate does have a result it will most likely mean that `n1`'s `branch` value will be larger than `fieldFlowLimit` (in the case of the C++ implementation, at least) even if we're only summing unique integers.

Reply:
I don't think it's complaining about the possible lack of result. The more detailed message says " An aggregate like 'sum' should work over a domain, otherwise duplicate values will not be counted. ". I think it wants you to say `sum(int i | i = getLanguageSpecificFlowIntoCallNodeCand1(_, n1, conf))`.

Reply:
That will also suffer from the same problem, though, right?
```ql
select sum([1, 1, 1, 1]), sum(int i | i = [1, 1, 1, 1])
```
both give `1`.

Reply:
After experimenting a bit to remind myself how `sum` works, I think it should be `sum(ParamNodeEx p1 | | getLanguageSpecificFlowIntoCallNodeCand1(p1, n1, conf))`.

Reply:
Oh, yes. That would fix the issue indeed! I'll go ahead and do that 🙇

Reply:
Fixed in 5068c315dd0cd4fd86b44fbc57dd3d43fb4649f7.

Reply:
I think it has to have three sections separated by `|`, like this:
```sum(ParamNodeEx p1 | | getLanguageSpecificFlowIntoCallNodeCand1(p1, n1, conf))```

Reply:
:doh: you're right! I've force pushed the right fix in c86494d91d1665a652db18932171641a36987e14 (and now I've actually done a `query compile` test to check the syntax!)